core/storage/suggestion/gae_models.py

```diff
@@ -152,93 +157,23 @@ def create(
             score_category=score_category).put()
 
     @classmethod
-    def get_suggestions_by_type(cls, suggestion_type):
-        """Gets all suggestions of a particular type.
-
-        Args:
-            suggestion_type: str. The type of the suggestions.
-
-        Returns:
-            list(SuggestionModel). A list of suggestions of the given
-                type, up to a maximum of feconf.DEFAULT_QUERY_LIMIT
-                suggestions.
-        """
-        return cls.get_all().filter(
-            cls.suggestion_type == suggestion_type).fetch(
-                feconf.DEFAULT_QUERY_LIMIT)
-
-    @classmethod
-    def get_suggestions_by_author(cls, author_id):
-        """Gets all suggestions created by the given author.
+    def query_suggestions(cls, queries):
+        """Queries for suggestions.
 
         Args:
-            author_id: str. The ID of the author of the suggestion.
+            queries: list(tuple). A list of queries. The first element in each
+                tuple is the field to be queried, and the second element is its
```

Comment:
... is the corresponding value to query for.

Reply:
Done